pkg/cmd/dedicated/register/registercluster.go

```diff
@@ -0,0 +1,350 @@
+package register
+
+import (
+	"context"
+	"fmt"
+	"github.com/AlecAivazis/survey/v2"
+	v1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+	dedicatedcmdutil "github.com/redhat-developer/app-services-cli/pkg/cmd/dedicated/dedicatedcmdutil"
+	kafkaFlagutil "github.com/redhat-developer/app-services-cli/pkg/cmd/kafka/flagutil"
+	kafkamgmtclient "github.com/redhat-developer/app-services-sdk-go/kafkamgmt/apiv1/client"
+
+	"github.com/redhat-developer/app-services-cli/pkg/shared/factory"
+	"github.com/spf13/cobra"
+	"strings"
+)
+
+type options struct {
+	selectedClusterId string
+	clusterList       v1.ClusterList
+	//interactive                   bool
+	selectedCluster         v1.Cluster
+	clusterMachinePoolList  v1.MachinePoolList
+	existingMachinePoolList []v1.MachinePool
+	//validatedMachinePoolList      []v1.MachinePool
+	selectedClusterMachinePool    v1.MachinePool
+	requestedMachinePoolNodeCount int
+	accessKafkasViaPrivateNetwork bool
+	newMachinePool                v1.MachinePool
+
+	f *factory.Factory
+}
+
+const (
+	machinePoolId           = "kafka-standard"
+	machinePoolTaintKey     = "bf2.org/kafkaInstanceProfileType"
+	machinePoolTaintEffect  = "NoExecute"
+	machinePoolTaintValue   = "standard"
+	machinePoolInstanceType = "m5.2xlarge"
+	machinePoolLabelKey     = "bf2.org/kafkaInstanceProfileType"
+	machinePoolLabelValue   = "standard"
+)
+
+func NewRegisterClusterCommand(f *factory.Factory) *cobra.Command {
+	opts := &options{
+		f: f,
+	}
+
+	//TODO add Localizer
+	cmd := &cobra.Command{
+		Use: "register-cluster",
+		// Short: f.Localizer.MustLocalize("registerCluster.cmd.shortDescription"),
+		Short:   "registerCluster.cmd.shortDescription",
+		Long:    "registerCluster.cmd.longDescription",
+		Example: "registerCluster.cmd.example",
+		Args:    cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, _ []string) error {
+			return runRegisterClusterCmd(opts)
+		},
+	}
+
+	// TODO add Localizer and logic for pass cluster ID flag
+	flags := kafkaFlagutil.NewFlagSet(cmd, f.Localizer)
+	// this flag will allow the user to pass the cluster id as a flag
+	opts.selectedClusterId = "123"
+	flags.StringVar(&opts.selectedClusterId, "cluster-id", "", "cluster id")
+	//flags.StringVar(&opts.selectedClusterId, "cluster-id", "", f.Localizer.MustLocalize("registerCluster.flag.clusterId"))
+
+	return cmd
+}
+
+func runRegisterClusterCmd(opts *options) error {
+	getListClusters(opts)
+	runClusterSelectionInteractivePrompt(opts)
+	getMachinePoolList(opts)
+	selectAccessPrivateNetworkInteractivePrompt(opts)
+	registerClusterWithKFM(opts)
+
+	return nil
+}
+
+func getListClusters(opts *options) error {
+	// ocm client connection
+	conn, err := opts.f.Connection()
```

Comment:
should we be closing the connection?

defer conn.Close()

Reply:
yes, the connection should def be closed but I am creating the connection in the [default client](pkg/shared/connection/api/defaultapi/default_client.go), I've already highlighted/asked what is the best way to close the connection? would simply putting a defer here, be the best way to do it?

Reply:
from the look of how you use the connection, you are not reusing the same connection but creating a connection for each ocm call.

In that regard, I'd just put a defer conn.Close() on each method that calls opts.f.Connection()